misc/toml-parser/src/main/java/io/ballerina/toml/validator/SchemaValidator.java

```diff
@@ -0,0 +1,232 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.toml.validator;
+
+import io.ballerina.toml.semantic.ast.TomlBooleanValueNode;
+import io.ballerina.toml.semantic.ast.TomlDoubleValueNodeNode;
+import io.ballerina.toml.semantic.ast.TomlKeyValueNode;
+import io.ballerina.toml.semantic.ast.TomlLongValueNode;
+import io.ballerina.toml.semantic.ast.TomlNode;
+import io.ballerina.toml.semantic.ast.TomlNodeVisitor;
+import io.ballerina.toml.semantic.ast.TomlStringValueNode;
+import io.ballerina.toml.semantic.ast.TomlTableArrayNode;
+import io.ballerina.toml.semantic.ast.TomlTableNode;
+import io.ballerina.toml.semantic.ast.TomlValueNode;
+import io.ballerina.toml.semantic.ast.TopLevelNode;
+import io.ballerina.toml.semantic.diagnostics.TomlDiagnostic;
+import io.ballerina.toml.semantic.diagnostics.TomlNodeLocation;
+import io.ballerina.toml.validator.schema.AbstractSchema;
+import io.ballerina.toml.validator.schema.ArraySchema;
+import io.ballerina.toml.validator.schema.NumericSchema;
+import io.ballerina.toml.validator.schema.ObjectSchema;
+import io.ballerina.toml.validator.schema.Schema;
+import io.ballerina.toml.validator.schema.StringSchema;
+import io.ballerina.toml.validator.schema.Type;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Visitor to validate toml object against rules in json schema.
+ *
+ * @since 2.0.0
+ */
+public class SchemaValidator extends TomlNodeVisitor {
+
+    private AbstractSchema schema;
+    private String key;
+
+    public SchemaValidator(Schema schema) {
+        this.schema = schema;
+    }
+
+    @Override
+    public void visit(TomlTableNode tomlTableNode) {
+        if (schema.type() != Type.OBJECT) {
+            TomlDiagnostic diagnostic = getTomlDiagnostic(tomlTableNode.location(), "TVE0002", "error.invalid.type",
+                    DiagnosticSeverity.ERROR, String.format("Key \"%s\" expects %s . Found object", this.key,
+                            schema.type()));
+            tomlTableNode.addDiagnostic(diagnostic);
+        } else {
+            ObjectSchema objectSchema = (ObjectSchema) schema;
+            Map<String, AbstractSchema> properties = objectSchema.properties();
+            Map<String, TopLevelNode> tableChildren = tomlTableNode.children();
```

Comment:
Usually a table has 'entries'

So better to change `tableChildren` -> `tableEntries`

It would also be better to change the method name as well `x.children()` to `x.entries()`